src/MagicSet.cpp

```diff
@@ -1161,7 +1161,7 @@ bool MagicSetTransformer::transform(AstTranslationUnit& translationUnit) {
                         newDirective->addKVP("filename", originalName.getNames()[0] + ".facts");
                     }
 
-                    newRelation->addLoad(std::unique_ptr<AstLoad>(newDirective));
+                    program->addLoad(std::unique_ptr<AstLoad>(newDirective));
```

Comment:
I'm not sure that this is needed, or if anything in MagicSet related to IO is needed.